### PR TITLE
#13064 implemented diag method for pytorch frontend

### PR DESCRIPTION
**Pull request by GSAUC3**

### CODE REVIEW

**Comment by GSAUC3** [comment]
Hi there, I tried implementing the pytorch, diag frontend function, the issue i am facing is that i can't test that function properly. as suggested by the video I have created a pull request, and would appreciate if anyone could review my code and help me out a bit. thank you.

**Comment by GSAUC3** [comment]
image

this is the error I am getting no matter what function I use for testing. I even tried testing on already merged functions like add or diagonal. even those functions are throwing the same error

File: ivy_tests/test_array_api
[the remark this answers is not in the document]
Comment (GSAUC3) [reply ivy_tests/test_array_api:1]: yes yes, sorry about that, I will revert the changes

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 1]
```diff
@@ -1230,3 +1230,48 @@ def test_torch_searchsorted(
...
+        )
+    )
+    return input_dtype,input_matrix
+
+@handle_frontend_test(
+    fn_tree="torch.diag",
+    dtype_and_x=_torch_diag_helper(),
+)
+def test_torch_diag(
+    dtype_and_x,
```
[the remark this answers is not in the document]
Comment (GSAUC3) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1263]: I am not sure if I am eligible to assign the PR, I didn't assign this one. I will try to do so if the option is available, else I will send the link to my new PR on discord, under pull request section tagging you. Will that be alright?

[the remark this answers is not in the document]
Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1263]: You could just tag me in a comment on the new PR and I will assign myself.

[the remark this answers is not in the document]
Comment (GSAUC3) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1264]: I tried cloning the repo and, but I am getting the same attribution error as before.

[the remark this answers is not in the document]
Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1264]: I copy-pasted your code to my local clone of the repo (in pycharm) and I don't get that error. I'm afraid there is not much else I can do to help on this. You could try asking on discord.

[the remark this answers is not in the document]
Comment (GSAUC3) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1264]: so does that mean the implementation is alright?

[the remark this answers is not in the document]
Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1264]: No, I get other errors that are related to mistakes in your code. You'll need to fix your setup to debug.

[the remark this answers is not in the document]
Comment (GSAUC3) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1264]: okay, I'll look into that.

**Pull request closed without merging**